src/lib/wallet/GoodWalletClass.js

```diff
@@ -363,7 +363,9 @@ export class GoodWallet {
         }
 
         const startBlock = this.lastEventsBlock
-        const lastBlock = await this.syncTxFromExplorer(startBlock).catch(e => {
+        const currentBlock = await this.getBlockNumber()
```

Comment:
why is this required? how does txs doesnt require the current block but this does?

Reply:
because there it's optional.